protocol/decoupled-hardfork-features.md

```diff
@@ -0,0 +1,308 @@
+  fjord_time = 1720627201 # Wed 10 Jul 2024 16:00:01 UTC
+  granite_time = 1726070401 # Wed 11 Sep 2024 16:00:01 UTC
+  holocene_time = 1736445601 # Thu 9 Jan 2025 18:00:01 UTC
+  isthmus_time = 1746806401 # Fri 9 May 2025 16:00:01 UTC
++ jovian_time = 20000000000 # This is automatically generated because superchain_time = 0. Custom chains can override it.
+
+ [features]
++ minimumBaseFee = "jovian" # This is automatically generated because superchain_time = 0.  Custom chains can override it.
++ configurableCallData = "jovian" # This is automatically generated because superchain_time = 0.  Custom chains can override it.
++ operatorFeeFix = "jovian" # This is automatically generated because superchain_time = 0.  Custom chains can override it.
+
+[optimism]
+  eip1559_elasticity = 6
+  eip1559_denominator = 50
+  eip1559_denominator_canyon = 250
+
+[genesis]
+  l2_time = 1686068903
+  [genesis.l1]
+    hash = "0x438335a20d98863a4c0c97999eb2481921ccd28553eac6f913af7c12aec04108"
+    number = 17422590
+  [genesis.l2]
+    hash = "0xdbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3"
+    number = 105235063
+  [genesis.system_config]
+    batcherAddress = "0x6887246668a3b87F54DeB3b94Ba47a6f63F32985"
+    overhead = "0x00000000000000000000000000000000000000000000000000000000000000bc"
+    scalar = "0x00000000000000000000000000000000000000000000000000000000000a6fe0"
+    gasLimit = 30000000
+
+[roles]
+  SystemConfigOwner = "0x847B5c174615B1B7fDF770882256e2D3E95b9D92"
+  ProxyAdminOwner = "0x5a0Aae59D09fccBdDb6C6CcEB07B7279367C3d2A"
+  Guardian = "0x09f7150D8c019BeF34450d6920f6B3608ceFdAf2"
+  Challenger = "0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A"
+  Proposer = "0x473300df21D047806A082244b417f96b32f13A33"
+  UnsafeBlockSigner = "0xAAAA45d9549EDA09E70937013520214382Ffc4A2"
+  BatchSubmitter = "0x6887246668a3b87F54DeB3b94Ba47a6f63F32985"
+
+[addresses]
+  AddressManager = "0xdE1FCfB0851916CA5101820A69b13a4E276bd81F"
+  L1CrossDomainMessengerProxy = "0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1"
+  L1ERC721BridgeProxy = "0x5a7749f83b81B301cAb5f48EB8516B986DAef23D"
+  L1StandardBridgeProxy = "0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1"
+  OptimismMintableERC20FactoryProxy = "0x75505a97BD334E7BD3C476893285569C4136Fa0F"
+  OptimismPortalProxy = "0xbEb5Fc579115071764c7423A4f12eDde41f106Ed"
+  SystemConfigProxy = "0x229047fed2591dbec1eF1118d64F7aF3dB9EB290"
+  ProxyAdmin = "0x543bA4AADBAb8f9025686Bd03993043599c6fB04"
+  AnchorStateRegistryProxy = "0x1c68ECfbf9C8B1E6C0677965b3B9Ecf9A104305b"
+  DelayedWETHProxy = "0x21429aF66058BC3e4aE4a8f2EC4531AaC433ecbC"
+  DisputeGameFactoryProxy = "0xe5965Ab5962eDc7477C8520243A95517CD252fA9"
+  FaultDisputeGame = "0x5738a876359b48A65d35482C93B43e2c1147B32B"
+  MIPS = "0xF027F4A985560fb13324e943edf55ad6F1d15Dc1"
+  PermissionedDisputeGame = "0x1Ae178eBFEECd51709432EA5f37845Da0414EdFe"
+  PreimageOracle = "0x1fb8cdFc6831fc866Ed9C51aF8817Da5c287aDD3"
+```
+
+# Alternatives Considered
```

Comment:
It's probably also worth listing the hard coded feature toggle I mentioned above as an option here. e.g.:

```
func IsOperatorFeeEnabled() {
  return IsIsthmusEnabled()
}
```
which could be replaced with `return false` if we need to disable the feature for some reason. Less flexible for testing, but no risk of misconfiguration. This approach is typically used on L1 though because which features are actually in a hard fork actually don't change very often.